// react-canvas-color-picker/dist/components/Handle/Handle.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const utils_1 = require("../../utils");
const Handle_module_css_1 = __importDefault(require("./Handle.module.css"));
function Handle(props) {
    const { spectrumFormat = "hsva", x, y, hue, saturation = 1, value = 1, alpha = 1, lightness = 0.5, className, } = props;
    const dpi = window.devicePixelRatio || 1;
    const bgColor = spectrumFormat === "hsla"
        ? {
            h: hue,
            s: saturation,
            l: lightness,
            a: alpha,
        }
        : utils_1.formatColorBySpectrum({
            h: hue,
            s: saturation,
            v: value,
            a: alpha,
        }, "hsla");
    return (React.createElement("div", { className: `${Handle_module_css_1.default.container} ${className}`, style: {
            transform: `translate(-50%, -50%) translate(${x / dpi}px, ${y / dpi}px)`,
        } },
        React.createElement("div", { className: Handle_module_css_1.default.handle, style: {
                backgroundColor: `hsla(${bgColor.h}, ${bgColor.s * 100}%, ${bgColor.l * 100}%, ${bgColor.a})`,
            } })));
}
exports.default = React.memo(Handle);
